### DISCUSSION TITLE
Process

### DISCUSSION THREAD

**Discussion by sturobson** [visual-framework/vf-core#1556] [General]
How do we go from "we need X" to it being available for use? What should you do if a component is _almost perfect_ but you need to add or change something slightly? What happens once a Pull Request is merged? At what point do we ask for feedback?

We already have a few pages on https://stable.visual-framework.dev that help understand how processes work a little. But there needs to be a larger document that 'covers everything'.

I was hoping to get this 'done' before finishing up - but I think where it is is probably best for you all to take on and improve. 

Here's the 'work-in-progress' –

…

# From ideation to release
## A components path to production.

~This is a working document to show the rough path a component goes from idea to production.~

The work is never done.

![Screenshot 2021-05-27 at 14 20 22](https://user-images.githubusercontent.com/925197/119840375-dd5fa100-befc-11eb-9495-9a21a75a8ed1.png)

### Idea

A new idea should start and be discussed in a GitHub discussion. 
An issue with an existing component not meeting requirements should be raised in a GitHub issue. 

### Design

Designs the components, with conversation with Development. Then iterates, if needed, on feedback from Development and stakeholder walkthroughs. 

### Development

Code the component,.
Iterate with Design.
Create a Pull Request.

### Release

Merge the Pull Request on Github.
- This updates the development CSS file (as used on latest.visual-framework.dev). 
Release to NPM. 
- This creates a ‘tagged’ release. So you can use all of the updated components from a single CSS file.
- This updates the individual components which you can choose to upgrade to or not.

**Comment by khawkins98** [comment]
Good stuff. As @sturobson mentions there are also a few pages on the docs that already get at this ... in fact we have quite a lot "stuff" on visual-framework.dev but we've not really fit it into the user journeys (our main target ones are: developers, page layout designers, stakeholders)

**Reply by khawkins98** [reply]
Addressed partly with https://github.com/visual-framework/vf-core/pull/1563

**Reply by khawkins98** [reply]
https://stable.visual-framework.dev/guidance/creating-components/
